src/nxt_var.c

```diff
@@ -132,25 +132,27 @@ nxt_var_ref_get(nxt_tstr_state_t *state, nxt_str_t *name)
 
     ref->index = state->var_refs->nelts - 1;
 
-    ref->name = nxt_str_dup(state->pool, NULL, name);
-    if (nxt_slow_path(ref->name == NULL)) {
-        return NULL;
-    }
-
     decl = nxt_var_hash_find(name);
 
     if (decl != NULL) {
         ref->handler = decl->handler;
         ref->cacheable = decl->cacheable;
 
-        return ref;
+        goto done;
     }
 
-    ret = nxt_http_unknown_var_ref(state, ref, name);
+    ret = nxt_http_unknown_var_ref(state->pool, ref, name);
     if (nxt_slow_path(ret != NXT_OK)) {
         return NULL;
     }
 
+done:
+
+    ref->name = nxt_str_dup(state->pool, NULL, name);
+    if (nxt_slow_path(ref->name == NULL)) {
+        return NULL;
+    }
+
     return ref;
 }
 
```

Comment:
I still can't help but feel this patch is doing too much.

The first change seems to be to pass in `state->pool` directly rather than just `state` to `nxt_http_unknown_var_ref()`

However most of the other changes seem to not be directly related to that. For example in the above hunk, we move the `ref->name = nxt_str_dup(state->pool, NULL, name);` to near the bottom of the function. This seems unrelated to the above change.

As well as the changes to  `nxt_var_interpreter()` and `nxt_var_cache_value()`.

To to recap, I think the API change `state` -> `state->pool` should be in its own preparatory commit.